doc/investigations/300-joint-ruptures/readme.md

```diff
@@ -0,0 +1,168 @@
+
+[MixedPlausibleClusterConnectionStrategy](..%2F..%2F..%2Fsrc%2Fmain%2Fjava%2Fnz%2Fcri%2Fgns%2FNZSHM22%2Fopensha%2Fruptures%2Fexperimental%2FMixedPlausibleClusterConnectionStrategy.java) is used to generate jumps between crustal clusters as normal. It intercepts building possible connections with subduction faults and will simply create the shortest jump between the two clusters (if within `maxJumpDist`) without doing any further vetting. This is due to performance reasons.
+
+We added some performance improvements to `opensha` but we have never been able to finish a rupture build due to the computational complexity.
+
+Kevin writes about this:
+
+```
+In theory the Coulomb filters should generally work in this environment,
+but the combinatorics would likely make it intractable (as you're
+finding out). If you do keep going down that route, there are a couple
+things to note that we might need to sort out:
+
+   - Some of the Coulomb filters are testing in every possible path
+     through the rupture, which is get tricky with down-dip subsectioning
+     (as you have on the subduction zone). We would have to be careful
+     that the prececessor/descendent relationships make sense at the
+     ClusterRupture.getTreeNavigator() level.
+
+   - Some of them assume uniform patch sizes (1km) on all surfaces, which
+     would be really inefficient when applied to an enormous subduction
+     source. You would probably want to discretized things at a different
+     resolution to make it run, which isn't yet selectable (but I can
+     make it so).
+
+Maybe a more sensible starting point would be to just take the crustal
+and subduction rupture sets and union them subject to some simpler
+Coulomb rules. I could help you design them, such as fraction of
+positive patch interactions filter  between the crustal and subduction
+sources. That could potentially be a decently quick calculation.
+```
+
+## Thinning
+
+With the sheer number of possible combinations of subduction and crustal ruptures, we need to consider thinning. The approach taken is to thin subduction and crustal ruptures before combining them into joint ruptures.
+
+For this investigation, the driver class is [ThinningReport](..%2F..%2F..%2Fsrc%2Fmain%2Fjava%2Fnz%2Fcri%2Fgns%2FNZSHM22%2Fopensha%2Fruptures%2Fexperimental%2Freports%2FThinningReport.java)
+
+Heavy lifting happens in [ThinningStats](..%2F..%2F..%2Fsrc%2Fmain%2Fjava%2Fnz%2Fcri%2Fgns%2FNZSHM22%2Fopensha%2Fruptures%2Fexperimental%2Freports%2FThinningStats.java)
+
+The code tentatively applies thinning and then creates a report to see the impact.
+
+Considerations:
+- Thinning should not change the size or magnitude distribution of ruptures
+- Thinning should not bias one region against another. Thinning needs to be applied evenly over NZ.
+
+Explored strategies:
+
+- Subduction filtering:
+    - size increase needs to be at least 10%
+    - ruptures of the same size avoid overlapping
+- Crustal ruptures:
```

Comment:
10 feels very restrictive. Can we open the aperture and still have a manageable number of possible ruptures?

Reply:
done

Reply:
(explained that I haven't explored this yet)